lib/modules/datasource/pypi/index.ts

```diff
@@ -34,45 +34,50 @@ export class PypiDatasource extends Datasource {
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    let dependency: ReleaseResult | null = null;
     // TODO: null check (#22198)
-    const hostUrl = ensureTrailingSlash(
-      registryUrl!.replace('https://pypi.org/simple', 'https://pypi.org/pypi'),
-    );
+    const hostUrl = ensureTrailingSlash(registryUrl!);
     const normalizedLookupName = PypiDatasource.normalizeName(packageName);
 
-    // not all simple indexes use this identifier, but most do
-    if (hostUrl.endsWith('/simple/') || hostUrl.endsWith('/+simple/')) {
+    const simpleHostUrl = hostUrl.replace(
+      'https://pypi.org/pypi',
+      'https://pypi.org/simple',
+    );
+    const simpleDependencies = await this.getResultsViaSimple(
```

Comment:
Is it possible that a package is only available via the JSON API?

If no, then we can return early.

Reply:
I think it is mostly about the metadata. As far as I can see, only `isPrivate` and `releases` may be set with the simple api

Reply:
What I mean is, if we get no result from Simple can we assume that JSON will also return nothing?

Reply:
Let's try both for maximum backwards compatibility